src/app/pages/course-add/course-add.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, ViewEncapsulation, OnInit, OnDestroy, ChangeDetectionStrategy } from '@angular/core';
+import { Router } from '@angular/router';
+import { NgForm } from '@angular/forms';
+
+import { BasePage } from '../base.page.component';
+import { ICourse } from './../../business-entities';
+import { CourseService, LoaderService } from './../../shared/services';
+
+@Component({
+	selector: 'course-add-page',
+	encapsulation: ViewEncapsulation.None,
+	providers: [],
+	styles: [require('./course-add.component.scss')],
+	template: require('./course-add.template.html'),
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CourseAddPageComponent extends BasePage {
+
+	private courseId: string;
+	private course: ICourse;
+
+	constructor(
+		private router: Router,
+		private courseService: CourseService,
+		private loaderService: LoaderService
+	) {
+		super();
+	}
+
+	onInit() {
+
+	}
+
+	onDestroy() {
+
+	}
+
+	public createCourse(courseForm: NgForm) {
+		this.loaderService.show();
+		let sub = this.courseService.create(courseForm.value)
+			.subscribe(
+				(resp) => {
+					console.log('course added', resp);
+					this.loaderService.hide();
+					this.router.navigate(['']);
```

Comment:
jesli this.loadService.hide() wystepuje i na completed i na error to nie ma potrzeby duplikacji kodu. Mozna uzyc operatora .do przed .subscribe:
.do(this.loadService.hide)
.subscribe...